pkg/resources/resource_source_table_postgres.go

```diff
@@ -0,0 +1,228 @@
+	"qualified_sql_name": QualifiedNameSchema("table"),
+	"source": IdentifierSchema(IdentifierSchemaParams{
+		Elem:        "source",
+		Description: "The source this table is created from.",
+		Required:    true,
+		ForceNew:    true,
+	}),
+	"upstream_name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "The name of the table in the upstream database.",
+	},
+	"upstream_schema_name": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		ForceNew:    true,
+		Description: "The schema of the table in the upstream database.",
+	},
+	"text_columns": {
+		Description: "Columns to be decoded as text.",
+		Type:        schema.TypeList,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		Optional:    true,
+		ForceNew:    true,
+	},
+	"comment":        CommentSchema(false),
+	"ownership_role": OwnershipRoleSchema(),
+	"region":         RegionSchema(),
+}
+
+func SourceTablePostgres() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: sourceTablePostgresCreate,
+		ReadContext:   sourceTablePostgresRead,
+		UpdateContext: sourceTablePostgresUpdate,
+		DeleteContext: sourceTableDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: sourceTablePostgresSchema,
+	}
+}
+
+func sourceTablePostgresCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	tableName := d.Get("name").(string)
+	schemaName := d.Get("schema_name").(string)
+	databaseName := d.Get("database_name").(string)
+
+	metaDb, region, err := utils.GetDBClientFromMeta(meta, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	o := materialize.MaterializeObject{ObjectType: "TABLE", Name: tableName, SchemaName: schemaName, DatabaseName: databaseName}
+	b := materialize.NewSourceTablePostgresBuilder(metaDb, o)
+
+	source := materialize.GetIdentifierSchemaStruct(d.Get("source"))
+	b.Source(source)
+
+	b.UpstreamName(d.Get("upstream_name").(string))
+
+	if v, ok := d.GetOk("upstream_schema_name"); ok {
+		b.UpstreamSchemaName(v.(string))
+	}
+
+	if v, ok := d.GetOk("text_columns"); ok {
+		textColumns, err := materialize.GetSliceValueString("text_columns", v.([]interface{}))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		b.TextColumns(textColumns)
+	}
+
+	if err := b.Create(); err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Handle ownership
+	if v, ok := d.GetOk("ownership_role"); ok {
+		ownership := materialize.NewOwnershipBuilder(metaDb, o)
+		if err := ownership.Alter(v.(string)); err != nil {
+			log.Printf("[DEBUG] resource failed ownership, dropping object: %s", o.Name)
+			b.Drop()
+			return diag.FromErr(err)
+		}
+	}
+
+	// Handle comments
+	if v, ok := d.GetOk("comment"); ok {
+		comment := materialize.NewCommentBuilder(metaDb, o)
+		if err := comment.Object(v.(string)); err != nil {
+			log.Printf("[DEBUG] resource failed comment, dropping object: %s", o.Name)
```

Comment:
Yep, those should be ok, they will only show if the `TF_LOG` env var is set to `DEBUG` and can be helpful for debugging, [similar to how other providers do this](https://github.com/search?q=repo%3ASnowflake-Labs%2Fterraform-provider-snowflake+%5BDEBUG%5D&type=code).

But this is actually a very good point, going through the TF docs, it looks like that they now suggest using their tflog package instead of the good old print + `[log_level]`. I've created a tracking issue for this so we could look into it further and possibly plan the work for this! https://github.com/MaterializeInc/terraform-provider-materialize/issues/664